### ISSUE TITLE
Spurious </div> after bibliography

### ISSUE THREAD

**Issue by mgeier** [mcmtroffaes/sphinxcontrib-bibtex#309]
This is maybe related to recent `docutils`/Sphinx changes?

In order to reproduce:

`conf.py`:

```
extensions = [
    'sphinxcontrib.bibtex',
]
bibtex_bibfiles = ['references.bib']
```

`index.rst`:

```
Hello!
======

citation: :cite:`perez2011python`

.. bibliography::

This is a paragraph after the bibliography.
```

`references.bib`:

```
@article{perez2011python,
author = {Pérez, Fernando and Granger, Brian E. and Hunter, John D.},
title = {Python: An Ecosystem for Scientific Computing},
journal = {Computing in Science Engineering},
volume = 13,
number = 2,
pages = {13--21},
doi = {10.1109/MCSE.2010.119},
year = 2011,
}
```

With those 3 files in the current directory, run:

    python3 -m sphinx . _build

This yields the following HTML:

```html
...
<p>citation: <span id="id1">[<a class="reference internal" href="#id3" title="Fernando Pérez, Brian E. Granger, and John D. Hunter. Python: an ecosystem for scientific computing. Computing in Science Engineering, 13(2):13–21, 2011. doi:10.1109/MCSE.2010.119.">PGH11</a>]</span></p>
<div class="docutils container" id="id2">
<div class="citation" id="id3" role="doc-biblioentry">
<span class="label"><span class="fn-bracket">[</span><a role="doc-backlink" href="#id1">PGH11</a><span class="fn-bracket">]</span></span>
<p>Fernando Pérez, Brian E. Granger, and John D. Hunter. Python: an ecosystem for scientific computing. <em>Computing in Science Engineering</em>, 13(2):13–21, 2011. <a class="reference external" href="https://doi.org/10.1109/MCSE.2010.119">doi:10.1109/MCSE.2010.119</a>.</p>
</div>
</div>
</div>
<p>This is a paragraph after the bibliography.</p>
...
```

The problem seems to be that there are two opening `<div ...>` but three closing `</div>`.

Versions:

* sphinxcontrib.bibtex: develop branch
* Sphinx: 5.x branch
* docutils: 0.19

**Comment by mcmtroffaes** [comment]
I assume you checked there is no matching preceding div? I'll see if I can reproduce.

**Comment by mcmtroffaes** [comment]
I've reproduced this now simply with the following one-line document:

```
.. [CIT2022] A citation.
```

When compiling this with sphinx, docutils fails to add an opening div for the citation list here:

https://github.com/docutils/docutils/blob/master/docutils/docutils/writers/_html_base.py#L632

It gets then closed here leading to the extra </div> tag:

https://github.com/docutils/docutils/blob/master/docutils/docutils/writers/_html_base.py#L640

Basically, this is due to odd behaviour of previous_sibling: if there is no previous sibling (i.e. index relative to parent is zero), it returns the node itself instead of None. I don't know if this is a bug or not. I'll report on the sphinx bug tracker.

**Comment by mcmtroffaes** [comment]
Docutils patch: https://sourceforge.net/p/docutils/patches/195/

**Comment by mgeier** [comment]
Thanks for the patch!

I can confirm that manually applying your patch fixes my original problem.

**Comment by mcmtroffaes** [comment]
Sweet, thanks for verifying from your end as well. Let's hope this makes it into docutils soon.

**Comment by mcmtroffaes** [comment]
Docutils patch merged upstream. Thanks again for reporting!

**Comment by choldgraf** [comment]
Did y'all figure out a way to fix this in a hacky way by any chance? Eg by monkey patching sphinx or something?.I'm trying to figure out if we are just stuck with docutils 17 until a new one comes out...

**Comment by mcmtroffaes** [comment]
No, the bug was traced quite deep into docutils...
